react/sagas/show_uri.js

```diff
@@ -12,45 +13,46 @@ function parseAndUpdateIdentifierAndClaim (modifier, claim) {
     ({ isChannel, channelName, channelClaimId, claimId } = lbryUri.parseIdentifier(modifier));
     ({ claimName, extension } = lbryUri.parseClaim(claim));
   } catch (error) {
-    return onRequestError(error.message);
+    return yield put(onRequestError(error.message));
   }
   // trigger an new action to update the store
   if (isChannel) {
-    return onNewAssetRequest(claimName, null, channelName, channelClaimId, extension);
-  } else {
-    return onNewAssetRequest(claimName, claimId, null, null, extension);
-  }
+    return yield put(onNewAssetRequest(claimName, null, channelName, channelClaimId, extension));
+  };
+  yield put(onNewAssetRequest(claimName, claimId, null, null, extension));
 }
-function parseAndUpdateClaimOnly (claim) {
+function* parseAndUpdateClaimOnly (claim) {
+  console.log('parseAndUpdateIdentifierAndClaim');
   // this could be a request for an asset or a channel page
   // claim could be an asset claim or a channel claim
   let isChannel, channelName, channelClaimId;
   try {
     ({ isChannel, channelName, channelClaimId } = lbryUri.parseIdentifier(claim));
   } catch (error) {
-    return onRequestError(error.message);
+    return yield put(onRequestError(error.message));
   }
   // trigger an new action to update the store
   // return early if this request is for a channel
   if (isChannel) {
-    return onNewChannelRequest(channelName, channelClaimId);
+    return yield put(onNewChannelRequest(channelName, channelClaimId));
```

Comment:
I think you need to add a state check here too. To see if that request exists. It looks like it will add the request to the list every time.